build-tools/packages/build-cli/src/filter.ts

```diff
@@ -212,7 +218,29 @@ export type PackageKind =
 /**
  * A convenience type mapping a directory containing a package to its PackageKind.
  */
-export interface PackageDetails {
-	package: Package;
-	kind: PackageKind;
+export class PackageDetails extends Package {
+	constructor(
+		private readonly pkg: string | Package,
+		public readonly group: string,
+		public readonly monoRepo: MonoRepo | undefined,
+		public readonly kind: PackageKind,
+	) {
+		super(
+			typeof pkg === "string" ? pkg : path.join(pkg.directory, "package.json"),
+			typeof pkg === "string" ? "none" : pkg.group,
+			typeof pkg === "string" ? undefined : pkg.monoRepo,
+		);
+		// let packageJson: string;
```

Comment:
Should these commented-out lines be removed? If not, probably worth adding a brief comment explaining why they're here.